pkg/apis/networkservice.mesh/v1/types.go

```diff
@@ -0,0 +1,129 @@
+// Copyright (c) 2018 Cisco and/or its affiliates.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at:
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package v1
+
+import (
+	meta_v1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+
+	"github.com/ligato/networkservicemesh/netmesh/model/netmesh"
+)
+
+// Constants to register CRDs for our resources
+const (
+	NSMGroup        string = "networkservice.mesh"
+	NSMGroupVersion string = "v1"
+
+	NSMEPPlural   string = "networkeservicendpoints"
+	FullNSMEPName string = NSMEPPlural + "." + NSMGroup
+
+	NSMChannelPlural   string = "networkservicechannels"
+	FullNSMChannelName string = NSMChannelPlural + "." + NSMGroup
+
+	NSMPlural   string = "networkservices"
+	FullNSMName string = NSMPlural + "." + NSMGroup
+)
+
+// Network Service Mesh Endpoint CRD
+// +genclient
+// +k8s:deepcopy-gen:interfaces=k8s.io/apimachinery/pkg/runtime.Object
+type NetworkServiceEndpoint struct {
+	meta_v1.TypeMeta   `json:",inline"`
+	meta_v1.ObjectMeta `json:"metadata,omitempty"`
+	Spec               netmesh.NetworkServiceEndpoint   `json:"spec"`
+	Validation         NetworkServiceEndpointValidation `json:"validation"`
+	Status             NetworkServiceEndpointStatus     `json:"status,omitempty"`
+}
+
+type NetworkServiceEndpointValidation struct {
+	Id   string `json:"^[a-fA-F0-9]{8}-[a-fA-F0-9]{4}-[a-fA-F0-9]{4}-[a-fA-F0-9]{4}-[a-fA-F0-9]{12}$"`
+	Name string `json:"^[a-fA-F0-9]{8}-[a-fA-F0-9]{4}-[a-fA-F0-9]{4}-[a-fA-F0-9]{4}-[a-fA-F0-9]{1,64}$"`
+}
+
+type NetworkServiceEndpointStatus struct {
+	State   string `json:"state,omitempty"`
+	Message string `kson:"message,omitempty"`
+}
+
+// -genclient
```

Comment:
The `List` types do not require clients, and in fact without the `// -genclient` tag, caused issues during compile.